src/gc.c

```diff
@@ -113,10 +179,34 @@ void GCContext_OnDelete(GCContext* gc) {
 }
 
 void GCContext_ForceInvoke(GCContext* gc, RedisModuleBlockedClient* bc) {
+  if (gc->isStopped == 1) return;
+
   BlockClients_push(&gc->bClients, bc);
-  RMUtilTimer_ForceInvoke(gc->timer);
+  GCContext_ForceBGInvoke(gc);
 }
 
 void GCContext_ForceBGInvoke(GCContext* gc) {
-  RMUtilTimer_Signal(gc->timer);
+  if (gc->isStopped == 1) return;
+
+  void* gcCtx;
+  RedisModuleCtx* ctx = RSDummyContext;
+  if (RedisModule_StopTimer(ctx, gc->timerID, &gcCtx) == REDISMODULE_OK) {
```

Comment:
Is there any condition in which the return value is not REDISMODULE_OK? also, why the assert?

Reply:
`RedisModule_StopTimer` returns ERR if no item was found in the queue.
All remove. Remains of previous tests.